File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        si = len(nums)
        p, flag = 1, 0
        for i in range(si):
            if nums[i] != 0:
                p *= nums[i]
            else:
                flag += 1
        res = []
        for i in range(si):
            if flag > 1:
                res.append(0)
            elif nums[i] != 0 and flag == 0:
                res.append(p // nums[i])
            else:
                if nums[i] != 0:
                    res.append(0 // p)
                else:
                    res.append(p)
        return res

File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int si=nums.size();
        int p=1,flag=0;
        for(int i=0;i<si;i++){
             if(nums[i]!=0)   
                p*=nums[i];
            else{
                flag++;
            }
        }
        vector<int>res;
        for(int i=0;i<si;i++){
            if(flag>1){
                res.emplace_back(0);
            }
            else if(nums[i]!=0 && flag==0){
                res.emplace_back(p/nums[i]);
            }
            else{
                if(nums[i]!=0){
                    res.emplace_back(0/p);
                }
                else{
                    res.emplace_back(p);
                }
            }
            
        }
        return res;
    }
};